pkg/global/apply_configuration.go

```diff
@@ -70,9 +42,20 @@ func ApplyConfiguration(configuration *pb.Configuration) error {
 			view.RegisterExporter(pe)
 		}
 
-		if tracingConfiguration.AlwaysSample {
-			trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+		sampling := float64(tracingConfiguration.SampleProbability)
+		if otelConfiguration := tracingConfiguration.GetOtel(); otelConfiguration != nil {
```

Comment:
Nit: Only use `.GetFoo()` methods in case the Protobuf message may be nil. In this case it's guaranteed to be non-nil.